<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Yiluan Song">
<meta name="dcterms.date" content="2023-10-04">

<title>Yiluan Song - Non-metric Multidimensional Scaling (NMDS)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../otter.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Yiluan Song</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../research.html"> 
<span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Blog</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
  <ul>
  <li><a href="#objectives" id="toc-objectives" class="nav-link active" data-scroll-target="#objectives">Objectives</a></li>
  <li><a href="#parametric-or-nonparametric" id="toc-parametric-or-nonparametric" class="nav-link" data-scroll-target="#parametric-or-nonparametric">Parametric or nonparametric?</a></li>
  <li><a href="#applications-of-nmds" id="toc-applications-of-nmds" class="nav-link" data-scroll-target="#applications-of-nmds">Applications of NMDS</a></li>
  <li><a href="#hands-on-community-composition-analysis" id="toc-hands-on-community-composition-analysis" class="nav-link" data-scroll-target="#hands-on-community-composition-analysis">Hands-on community composition analysis</a></li>
  <li><a href="#permanova" id="toc-permanova" class="nav-link" data-scroll-target="#permanova">PERMANOVA</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<div class="quarto-about-solana column-body content">
  <div class="about-entity">
    <div class="entity-contents">
      <header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Non-metric Multidimensional Scaling (NMDS)</h1>
  <div class="quarto-categories">
    <div class="quarto-category">Statistics</div>
  </div>
  </div>
<div class="quarto-title-meta">
    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Yiluan Song </p>
          </div>
  </div>
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 4, 2023</p>
    </div>
  </div>
  </div>
</header> 
      <div class="about-contents"><main class="content" id="quarto-document-content">

<p>Guest lecture for Dr.&nbsp;Drew Gronewold’s class EAS 501.077 Multivariate Statistics for Environmental Science in 2023 Fall.</p>
<section id="objectives" class="level2">
<h2 data-anchor-id="objectives">Objectives</h2>
<ul>
<li>Compare parametric and nonparametric methods for ordination</li>
<li>Learn about real-life applications of NMDS</li>
<li>Practice using NMDS in R to analyze community composition</li>
<li>Get familiar with a statistical test, PERMANOVA</li>
</ul>
</section>
<section id="parametric-or-nonparametric" class="level2">
<h2 data-anchor-id="parametric-or-nonparametric">Parametric or nonparametric?</h2>
<p>In previous classes, we have learned about principal component analysis (PCA).</p>
<p>PCA have certain assumptions, requiring the data to be * continuous, * linear, * normally distributed.</p>
<p>Note that PCA involves calculating the Euclidean distances in the multidimensional space between samples.</p>
<p>What if our data are * discrete (e.g., presence/absence, category), * nonlinear (e.g., day of year, precipitation), * not normally distributed (e.g., count of rare species)?</p>
<p>Does Euclidean distance still make sense?</p>
<blockquote class="blockquote">
<p>“Nonparametric statistics is the type of statistics that is not restricted by assumptions concerning the nature of the population from which a sample is drawn.”</p>
</blockquote>
<p>Quote from <a href="https://en.wikipedia.org/wiki/Nonparametric_statistics">Wikipedia</a>.</p>
<p>PCA is a <em>parametric</em> method for ordination. Today, we are going to add a <em>nonparametric</em> method into our toolbox: non-metric multidimensional scaling (NMDS).</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="600"></p>
<figcaption>Example NMDS plot.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://jonlefcheck.net/2012/10/24/nmds-tutorial-in-r/">Source</a></p>
<p>NMDS places samples that are more “similar” to each other closer in a low-dimensional space.</p>
<p>Advantages of NMDS include * accommodating multiple types of data, * being used with different measures of dissimilarity.</p>
<p>However, be aware of the differences with PCA. Unlike principle components in PCA, the axes in NMDS do not carry a specific meaning.</p>
</section>
<section id="applications-of-nmds" class="level2">
<h2 data-anchor-id="applications-of-nmds">Applications of NMDS</h2>
<p>Case 1: Phytoplankton community composition (<a href="https://www.sciencedirect.com/science/article/pii/S0925857418304749">Paper</a>)</p>
<p>Tropical urban reservoirs face the problem of phytoplankton bloom, often dominated by toxic cyanobacteria. It has been hypothesized that macrophytes (water plants) can inhibit phytoplankton, presenting an opportunity for reservoir restoration.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="624"></p>
<figcaption>Cyanobacterial blooms.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://www.nature.com/articles/s41579-018-0040-1">Huisman et al.&nbsp;(2019)</a></p>
<p>In reservoirs in Singapore, we conducted a series of experiments to test macrophytes’ ability to alter phytoplankton communities and facilitate reservoir restoration.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="932"></p>
<figcaption>Set up of a mesocosm experiment.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://link.springer.com/article/10.1007/s10750-021-04679-7">Sim et al.&nbsp;(2021)</a> Fig. 1</p>
<p>Apart from the effect of macrophytes on phytoplankton biomass, we also care about the effects on phytoplankton community composition.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="696"></p>
<figcaption>Relative abundance (biovolume) of phytoplankton taxa.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://link.springer.com/article/10.1007/s10750-021-04679-7">Sim et al.&nbsp;(2021)</a> Fig. 3</p>
<p>Community composition data is essentially multivariate. The (relative) abundance of each taxa is a variable, and there are usually many taxa. How do we analyze the response of all these variables together?</p>
<p>We used NMDS to visualize the effect of macrophyte treatment on phytoplankton community composition.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="756"></p>
<figcaption>NMDS of the relative biovolumes of phytoplankton genera in the control and macrophyte treatments.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://www.sciencedirect.com/science/article/pii/S0925857418304749">Mowe et al.&nbsp;(2019)</a> Fig. 6</p>
<p>Here we see how the introduction of macrophyte caused a shift in phytoplankton community composition, with the shift being greater as density of macrophyte increased.</p>
<p>Case 2: Grassland taxonomic, phylogenitic, and functional trait composition (<a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/1365-2745.13808">Paper</a>)</p>
<p>The trajectory of early succession communities is shaped by the plant phylogenetic and trait history. Teasing apart these two processes has important implications for restoration.</p>
<p>Holding starting species richness constant, Karimi et al.&nbsp;planted communities of different phylogenetic diversity (PD) and functional trait diversity (FD).</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="1064"></p>
<figcaption>NMDS of taxonomic, phylogenetic and functional trait composition in two types of restoration treatments.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/1365-2745.13808">Karimi et al.&nbsp;(2021)</a> Fig. 4</p>
<p>Similar to case study 1, they examined how different treatments drive differences in community composition. Apart from taxonomic composition, they also analyzed responses in phylogenetic and functional trait composition.</p>
<p>I highlight this study because they used a combination of data types to characterize composition.</p>
<blockquote class="blockquote">
<p>“Functional diversity was assessed using 12 continuous leaf traits, 6 categorical traits, 8 binary root traits, seed mass, a categorical habitat moisture trait and genome size.”</p>
</blockquote>
<p>They also used NMDS to study changes in the three kinds of community composition.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="1064"></p>
<figcaption>NMDS of taxonomic, phylogenetic and functional trait composition before and after restoration treatment.</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/1365-2745.13808">Karimi et al.&nbsp;(2021)</a> Fig. 4 (Note their dissimilarity measures.)</p>
<p>Here, we see that the PD treatment increased the dispersion of taxonomic and phylogenetic composition between communities (increased beta diversity) and caused directional shifts in functional trait composition (convergence).</p>
</section>
<section id="hands-on-community-composition-analysis" class="level2">
<h2 data-anchor-id="hands-on-community-composition-analysis">Hands-on community composition analysis</h2>
<p>There are really good NMDS <a href="https://jonlefcheck.net/2012/10/24/nmds-tutorial-in-r/">tutorial1</a> <a href="https://rpubs.com/CPEL/NMDS">tutorial 2</a> which I encourage you to try at home. Here I give another example with real-life data and some other visualization options.</p>
<blockquote class="blockquote">
<p>Background: The National Ecological Observatory Network (NEON) collects ecological and biogeochemical data with standardized protocols across 81 field sites across the United States. The Woody Plant Vegetation Structure dataset (DP1.10098) describes the structure and composition of woody vegetation through the mapping, identification, and measurement of free-standing woody plants including trees, saplings, shrubs, lianas, etc.</p>
</blockquote>
<p>We have downloaded NEON vegetation structure data at one site, Bartlett Experimental Forest (BART), for you to analyze the composition of woody vegetation. Please download them <a href="https://github.com/yiluansong/teaching/blob/main/content/post/2023-10-04-nmds/data.rds">here</a>.</p>
<p>Read in some data frames.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="st">"data.rds"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>df_tree <span class="ot">&lt;-</span> dat<span class="sc">$</span>vst_apparentindividual <span class="sc">%&gt;%</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(publicationDate)) <span class="sc">%&gt;%</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(plotID, individualID, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(plantStatus <span class="sc">%&gt;%</span> <span class="fu">tolower</span>(), <span class="st">"live"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(plotID, individualID)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>df_tree_sp <span class="ot">&lt;-</span> dat<span class="sc">$</span>vst_mappingandtagging <span class="sc">%&gt;%</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(publicationDate)) <span class="sc">%&gt;%</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(individualID, plotID, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(individualID, plotID, taxonID, scientificName) <span class="sc">%&gt;%</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(taxonID)) <span class="sc">%&gt;%</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxonID <span class="sc">!=</span> <span class="st">"2PLANT"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>df_plot <span class="ot">&lt;-</span> dat<span class="sc">$</span>vst_perplotperyear <span class="sc">%&gt;%</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(publicationDate)) <span class="sc">%&gt;%</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(plotID, plotType, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(plotID, plotType, nlcdClass, <span class="at">lon =</span> decimalLongitude, <span class="at">lat =</span> decimalLatitude)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Join the data frames. Each row is an individual tree (with unique individualID). Feel free to explore this dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df_tree <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_tree_sp,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"individualID"</span>, <span class="st">"plotID"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_plot,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"plotID"</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     plotID            individualID taxonID                 scientificName
1  BART_075 NEON.PLA.D01.BART.03230    FAGR        Fagus grandifolia Ehrh.
2  BART_075 NEON.PLA.D01.BART.04011    FAGR        Fagus grandifolia Ehrh.
3  BART_036 NEON.PLA.D01.BART.04548    FAGR        Fagus grandifolia Ehrh.
4  BART_075 NEON.PLA.D01.BART.04702    TSCA Tsuga canadensis (L.) Carrière
5  BART_075 NEON.PLA.D01.BART.03257    FAGR        Fagus grandifolia Ehrh.
6  BART_075 NEON.PLA.D01.BART.03148    FAGR        Fagus grandifolia Ehrh.
7  BART_075 NEON.PLA.D01.BART.03254    FAGR        Fagus grandifolia Ehrh.
8  BART_075 NEON.PLA.D01.BART.04703    ACPE          Acer pensylvanicum L.
9  BART_036 NEON.PLA.D01.BART.04554    FAGR        Fagus grandifolia Ehrh.
10 BART_075 NEON.PLA.D01.BART.03300   BEAL2  Betula alleghaniensis Britton
   plotType       nlcdClass       lon      lat
1     tower     mixedForest -71.28649 44.05914
2     tower     mixedForest -71.28649 44.05914
3     tower deciduousForest -71.28588 44.06208
4     tower     mixedForest -71.28649 44.05914
5     tower     mixedForest -71.28649 44.05914
6     tower     mixedForest -71.28649 44.05914
7     tower     mixedForest -71.28649 44.05914
8     tower     mixedForest -71.28649 44.05914
9     tower deciduousForest -71.28588 44.06208
10    tower     mixedForest -71.28649 44.05914</code></pre>
</div>
</div>
<p>Process the joined data frame to get a community composition data frame. Each row is a community (a plot). Species names are now columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>df_comm <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(plotID, plotType, nlcdClass, lat, lon, scientificName) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(<span class="at">key =</span> <span class="st">"scientificName"</span>, <span class="at">value =</span> <span class="st">"count"</span>, <span class="at">fill =</span> <span class="dv">0</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>df_comm <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 10 × 36
   plotID   plotType    nlcdClass   lat   lon `Abies sp.` Acer pensylvanicum L…¹
   &lt;chr&gt;    &lt;chr&gt;       &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;       &lt;dbl&gt;                  &lt;dbl&gt;
 1 BART_001 distributed mixedFor…  44.0 -71.3           0                      2
 2 BART_002 distributed deciduou…  44.0 -71.3           0                      6
 3 BART_003 distributed deciduou…  44.1 -71.3           0                      0
 4 BART_004 distributed mixedFor…  44.0 -71.3           0                      1
 5 BART_005 distributed mixedFor…  44.1 -71.3           1                      0
 6 BART_006 distributed deciduou…  44.1 -71.3           1                      6
 7 BART_007 distributed mixedFor…  44.0 -71.3           0                      0
 8 BART_010 distributed deciduou…  44.1 -71.3           0                      0
 9 BART_011 distributed mixedFor…  44.1 -71.3           0                      0
10 BART_012 distributed deciduou…  44.0 -71.3           0                      0
# ℹ abbreviated name: ¹​`Acer pensylvanicum L.`
# ℹ 29 more variables: `Acer rubrum L.` &lt;dbl&gt;, `Acer saccharinum L.` &lt;dbl&gt;,
#   `Acer saccharum Marshall` &lt;dbl&gt;,
#   `Acer saccharum Marshall var. saccharum` &lt;dbl&gt;, `Acer sp.` &lt;dbl&gt;,
#   `Betula ×caerulea Blanch. var. caerulea` &lt;dbl&gt;,
#   `Betula alleghaniensis Britton` &lt;dbl&gt;, `Betula lenta L.` &lt;dbl&gt;,
#   `Betula papyrifera Marshall` &lt;dbl&gt;, …</code></pre>
</div>
</div>
<p>We need to make this community composition data frame a matrix. Again, each row is a community (a plot) and each species is a column. Note that we leave out the metadata for plots.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>mat_comm <span class="ot">&lt;-</span> df_comm <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>plotID, <span class="sc">-</span>plotType, <span class="sc">-</span>nlcdClass, <span class="sc">-</span>lon, <span class="sc">-</span>lat) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>mat_comm[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     Abies sp. Acer pensylvanicum L. Acer rubrum L. Acer saccharinum L.
[1,]         0                     2              1                   0
[2,]         0                     6              0                   0
[3,]         0                     0             15                   0
[4,]         0                     1              5                   0
[5,]         1                     0              0                   0
[6,]         1                     6              0                   0
     Acer saccharum Marshall Acer saccharum Marshall var. saccharum
[1,]                       0                                      1
[2,]                       0                                     22
[3,]                       0                                      7
[4,]                       0                                      0
[5,]                       0                                      0
[6,]                       0                                      2</code></pre>
</div>
</div>
<p>With this community composition matrix, we can use the <em>metaMDS</em> function in <em>vegan</em> package to perform NMDS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>mds_comm <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">metaMDS</span>(mat_comm, <span class="at">distant =</span> <span class="st">"bray"</span>, <span class="at">k =</span> <span class="dv">4</span>, <span class="at">try =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Square root transformation
Wisconsin double standardization
Run 0 stress 0.08885463 
Run 1 stress 0.09016598 
Run 2 stress 0.08885805 
... Procrustes: rmse 0.00397431  max resid 0.01364567 
Run 3 stress 0.08885508 
... Procrustes: rmse 0.0001215631  max resid 0.0004105488 
... Similar to previous best
Run 4 stress 0.08885586 
... Procrustes: rmse 0.0003292734  max resid 0.001099849 
... Similar to previous best
Run 5 stress 0.0903252 
Run 6 stress 0.08885492 
... Procrustes: rmse 0.0001005969  max resid 0.0003412662 
... Similar to previous best
Run 7 stress 0.08885847 
... Procrustes: rmse 0.004096719  max resid 0.01408031 
Run 8 stress 0.08885396 
... New best solution
... Procrustes: rmse 0.002743248  max resid 0.00947331 
... Similar to previous best
Run 9 stress 0.09029896 
Run 10 stress 0.09034585 
Run 11 stress 0.08893074 
... Procrustes: rmse 0.009504952  max resid 0.03165667 
Run 12 stress 0.09032054 
Run 13 stress 0.08885345 
... New best solution
... Procrustes: rmse 0.002182506  max resid 0.007588686 
... Similar to previous best
Run 14 stress 0.08889528 
... Procrustes: rmse 0.005338802  max resid 0.01761426 
Run 15 stress 0.08886069 
... Procrustes: rmse 0.003999398  max resid 0.01384676 
Run 16 stress 0.08885448 
... Procrustes: rmse 0.0005095852  max resid 0.001713473 
... Similar to previous best
Run 17 stress 0.09028926 
Run 18 stress 0.08886793 
... Procrustes: rmse 0.005178257  max resid 0.0181378 
Run 19 stress 0.08885554 
... Procrustes: rmse 0.002771664  max resid 0.009498437 
... Similar to previous best
Run 20 stress 0.09016291 
*** Best solution repeated 3 times</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>mds_comm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
vegan::metaMDS(comm = mat_comm, k = 4, try = 100, distant = "bray") 
global Multidimensional Scaling using monoMDS
Data:     wisconsin(sqrt(mat_comm)) 
Distance: bray 
Dimensions: 4 
Stress:     0.08885345 
Stress type 1, weak ties
Best solution was repeated 3 times in 20 tries
The best solution was from try 13 (random start)
Scaling: centring, PC rotation, halfchange scaling 
Species: expanded scores based on 'wisconsin(sqrt(mat_comm))' </code></pre>
</div>
</div>
<p>We check the stress value to find out if our NMDS has a good fit. We can also make a stressplot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>mds_comm<span class="sc">$</span>stress</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.08885345</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">stressplot</span>(mds_comm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>NMDS stress plot.</figcaption>
</figure>
</div>
</div>
</div>
<p>Is our stress value considered good? A rule of thumb is that stress &lt; 0.1 is good and stress &lt; 0.05 is excellent. You can increase the number of dimensions (k) to reduce stress. However, large k is usually not useful and can even be harmful.</p>
<p>For more considerations, read this <a href="https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/nmds/">chapter</a>.</p>
<p>Now we can generate a basic NMDS plot. Labels in black show communities, and labels in red show species.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mds_comm, <span class="at">type =</span> <span class="st">"t"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>A basic NMDS plot.</figcaption>
</figure>
</div>
</div>
</div>
<p>You can see that some communities are more similar than others, and some species tend to occur together.</p>
<p>We can try to redraw the NMDS plot using <em>ggplot</em>. This gives you more control on the graph elements.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>df_nmds_comm <span class="ot">&lt;-</span> mds_comm <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>(<span class="at">display =</span> <span class="st">"sites"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(df_comm <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(plotID, plotType, nlcdClass))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>df_nmds_sp <span class="ot">&lt;-</span> mds_comm <span class="sc">%&gt;%</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">scores</span>(<span class="at">display =</span> <span class="st">"species"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"species"</span>)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_nmds_comm, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2)) <span class="sc">+</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> df_nmds_sp, <span class="fu">aes</span>(NMDS1, NMDS2, <span class="at">label =</span> species), <span class="at">color =</span> <span class="st">"dark grey"</span>) <span class="sc">+</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_few</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>A NMDS plot drawn using ggplot.</figcaption>
</figure>
</div>
</div>
</div>
<p>We ca draw ellipses based on existing grouping of these communities. In experiment, we can draw ellipses for control and treatment. In observations, we can draw ellipses for different time points. Here, I draw ellipses for communities from different land cover types.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_nmds_comm, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">col =</span> nlcdClass)) <span class="sc">+</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> df_nmds_sp, <span class="fu">aes</span>(NMDS1, NMDS2, <span class="at">label =</span> species), <span class="at">color =</span> <span class="st">"dark grey"</span>) <span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_few</span>() <span class="sc">+</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>NMDS plot showing composition of communities from different land cover types.</figcaption>
</figure>
</div>
</div>
</div>
<p>You can see some differences in the composition of communities from different land cover types. Note that we used the first two axes of NMDS. What if we use another two axes?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_nmds_comm, <span class="fu">aes</span>(<span class="at">x =</span> NMDS3, <span class="at">y =</span> NMDS4, <span class="at">col =</span> nlcdClass)) <span class="sc">+</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> df_nmds_sp, <span class="fu">aes</span>(NMDS1, NMDS2, <span class="at">label =</span> species), <span class="at">color =</span> <span class="st">"dark grey"</span>) <span class="sc">+</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_few</span>() <span class="sc">+</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>NMDS plot showing composition of communities from different land cover types, using NMDS3 and NMDS4.</figcaption>
</figure>
</div>
</div>
</div>
<p>We still see some differences using NMDS3 and NMDS4, but perhaps less distinct compared to when we used NMDS1 and NMDS2.</p>
<p>NEON has two types of plots, distributed and tower. Their sampling methods differ. Let’s see their difference.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df_nmds_comm, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">col =</span> plotType)) <span class="sc">+</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> df_nmds_sp, <span class="fu">aes</span>(NMDS1, NMDS2, <span class="at">label =</span> species), <span class="at">color =</span> <span class="st">"dark grey"</span>) <span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_few</span>() <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>NMDS plot showing composition of communities from different plot types.</figcaption>
</figure>
</div>
</div>
</div>
<p>Communities from distributed plots seem to be more dispersed? The reason might be tower plots have a more constrained sampling area.</p>
</section>
<section id="permanova" class="level2">
<h2 data-anchor-id="permanova">PERMANOVA</h2>
<p>We have done some visualization that hopefully help us intuitively see the similarity and differences between groups of samples. What if we are asked to statistically quantify the differences between these groups? How can we get a <em>p</em> value?</p>
<p>Permutational multivariate analysis of variance (PERMANOVA) is a nonparametric multivariate statistical permutation test. * A significant <em>p</em> value indicates that the two groups are different in the their centroids OR dispersion in the multidimensional space. * It is similar to ANOVA, but it does not have many assumptions except exchangeability (usually satisfied).</p>
<p>(The two case studies we introduced both used PERMANOVA.)</p>
<p>In practice, we can easily use the <em>adonis2</em> function from the <em>vegan</em> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>res_permanova <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(mat_comm <span class="sc">~</span> nlcdClass, <span class="at">data =</span> df_comm, <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>res_permanova</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999
vegan::adonis2(formula = mat_comm ~ nlcdClass, data = df_comm, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     2   2.1141 0.31126 8.3605  1e-04 ***
Residual 37   4.6781 0.68874                  
Total    39   6.7922 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>res_permanova <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(mat_comm <span class="sc">~</span> plotType, <span class="at">data =</span> df_comm, <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>res_permanova</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999
vegan::adonis2(formula = mat_comm ~ plotType, data = df_comm, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     1   1.0483 0.15434 6.9352  6e-04 ***
Residual 38   5.7439 0.84566                  
Total    39   6.7922 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>res_permanova <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(mat_comm <span class="sc">~</span> nlcdClass <span class="sc">*</span> plotType, <span class="at">data =</span> df_comm, <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>res_permanova</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 9999
vegan::adonis2(formula = mat_comm ~ nlcdClass * plotType, data = df_comm, permutations = 9999)
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     4   3.1207 0.45946 7.4374  1e-04 ***
Residual 35   3.6715 0.54054                  
Total    39   6.7922 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Can you try to interpret the results of these three PERMANOVA test that correspond to the three NMDS plots above? Are the differences that we observed from NMDS plots statistically significant?</p>


</section>
</main></div>
    </div>
  </div>
</div>
 <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/yiluansong\.github\.io\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>